nexus/db-model/src/dataset.rs

```diff
@@ -81,6 +93,59 @@ impl Dataset {
     }
 }
 
+impl From<BlueprintDatasetConfig> for Dataset {
+    fn from(bp: BlueprintDatasetConfig) -> Self {
+        let kind = DatasetKind::from(&bp.kind);
+        let (size_used, zone_name) = match bp.kind {
+            ApiDatasetKind::Crucible => (Some(0), None),
+            ApiDatasetKind::Zone { name } => (None, Some(name)),
+            _ => (None, None),
```

Comment:
Both done in fdab22a75c794bbddbc8d550008a8b1848a9e347 (rename and full match).

We already have a `zone_name` method to return `Option<&str>`, so I'll use that. As for the `size_used` value, it's really only used by Crucible, but I can add an exhaustive match to avoid `_`. That said, I think we'll only be matching on Crucible for now.